src/rpc/mempool.cpp

```diff
@@ -587,6 +587,89 @@ static RPCHelpMan getmempoolentry()
     };
 }
 
+static RPCHelpMan getmempooltxspendingprevout()
```

Comment:
Also a nit (especially since that PR didn't exist when you opened this one): if we end up with #24539 as well, a more generic RPC named `gettxspendingprevout` might be better, i.e. to retrieve a transaction from block or from mempool.

Reply:
Good idea! I changed that in https://github.com/bitcoin/bitcoin/pull/24408/commits/8c9152c1098acec4b1b4e333240f0e95eea21159 so that #24539 can build on top of it. I currently kept it in `mempool.cpp` and in the mempool RPCs list, I feel it's the responsibility of #24539 to move it elsewhere, is that ok @sstone?

I think #24539 won't have to modify any of the RPC fields, which is great. Callers will simply follow up with `getrawtransaction` which will provide the `confirmations` field, informing them of whether the sending transaction is in the mempool or is confirmed.

Reply:
Yes, and I agree that it makes sense to rebase https://github.com/bitcoin/bitcoin/pull/24539 on this PR